packages/amplify-category-api/src/graphql-transformer/transformer-options-v2.ts

```diff
@@ -0,0 +1,302 @@
+import {
+  $TSAny,
+  $TSContext, $TSMeta, $TSObject,
+  AmplifyCategories,
+  AmplifySupportedService, ApiCategoryFacade,
+  CloudformationProviderFacade, getGraphQLTransformerAuthDocLink,
+  JSONUtilities,
+  pathManager,
+  stateManager,
+} from 'amplify-cli-core';
+import { AppSyncAuthConfiguration } from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  collectDirectivesByTypeNames,
+} from '@aws-amplify/graphql-transformer-core';
+import {
+  getSanityCheckRules,
+  loadProject,
+} from 'graphql-transformer-core';
+import path from 'path';
+import fs from 'fs-extra';
+import { ResourceConstants } from 'graphql-transformer-common';
+import _ from 'lodash';
+import { getAdminRoles, getIdentityPoolId } from './utils';
+import { schemaHasSandboxModeEnabled, showSandboxModePrompts } from './sandbox-mode-helpers';
+import { getTransformerFactory } from './transformer-factory';
+import { AmplifyCLIFeatureFlagAdapter } from './amplify-cli-feature-flag-adapter';
+import {
+  DESTRUCTIVE_UPDATES_FLAG,
+  PARAMETERS_FILENAME,
+  PROVIDER_NAME,
+  ROOT_APPSYNC_S3_KEY,
+} from './constants';
+import {
+  TransformerFactoryArgs,
+  TransformerProjectOptions,
+} from './transformer-options-types';
+import { contextUtil } from '../category-utils/context-util';
+
+export const APPSYNC_RESOURCE_SERVICE = 'AppSync';
+
+/**
+ * Use current context to generate the Transformer options for generating
+ * a GraphQL Transformer V2 object
+ * @param context The $TSContext from the Amplify CLI
+ * @param options The $TSAny options config coming from the Amplify CLI
+ */
+export const generateTransformerOptions = async (
```

Comment:
Is this logic cloning in from somewhere? This reads like it's a dup of something, but it's not obvious to me where we're copying it from, and how we're keeping them in-sync until the migration is finished.

Reply:
Yeah this is a bunch of extracted logic from the transform schema method for V2, to be able to produce a transformer we have to get all this data together and we have to be able to instantiate a transformer object for the single source read. I'll look at the refactor again, maybe I can just get this to work

Reply:
Update: I did refactor it and got E2E to pass this time